lcls-twincat-motion/Library/POUs/Motion/States/FB_PositionStateMoveND.TcPOU

```diff
@@ -0,0 +1,139 @@
+END_VAR
+VAR_INPUT
+    // The number of motors we're actually using
+    nActiveMotorCount: UINT;
+    // Start all moves on rising edge, stop all moves on falling edge
+    bExecute: BOOL;
+    // Reset any errors
+    bReset: BOOL;
+    // Define behavior for when a move request is interrupted with a new request
+    enumMotionRequest: ENUM_MotionRequest := ENUM_MotionRequest.WAIT;
+END_VAR
+VAR_OUTPUT
+    // TRUE if ALL of the motors are at their goal states
+    bAtState: BOOL;
+    // TRUE if ANY of this FB's moves are in progress
+    bBusy: BOOL;
+    // TRUE if ALL motors have completed the last move request from this FB
+    bDone: BOOL;
+    // TRUE if ANY of this FB's moves had an error
+    bError: BOOL;
+    // How many FBs are erroring
+    nErrorCount: UINT;
+    // Which component is the source of the example/summarized error
+    nShownError: DINT;
+    // One of the error ids
+    nErrorID: UDINT;
+    // The error error message matching the ID
+    sErrorMessage: STRING;
+END_VAR
+VAR
+    // 1D State movers: FBs to move the motors
+    afbPositionStateMove: ARRAY[1..MotionConstants.MAX_STATE_MOTORS] OF FB_PositionStateMove;
+    nIndex: DINT;
+    bMotorCountError: BOOL;
+    nLowerBound: DINT;
+    nUpperBound: DINT;
+END_VAR
+]]></Declaration>
+    <Implementation>
+      <ST><![CDATA[
+CheckCount();
+IF NOT bMotorCountError THEN
+    DoStateMoves();
+    CombineOutputs();
+END_IF
+]]></ST>
+    </Implementation>
+    <Action Name="CheckCount" Id="{9244b3c0-fac6-4a82-83fe-d3ec9dc81390}">
+      <Implementation>
+        <ST><![CDATA[
+// Make sure the motor count is valid (positive, nonzero, less or equal to the max)
+bMotorCountError S= nActiveMotorCount <= 0;
+bMotorCountError S= nActiveMotorCount > MotionConstants.MAX_STATE_MOTORS;
+
+IF bMotorCountError THEN
+    bError := TRUE;
+    sErrorMessage := 'Internal Error: invalid motor count';
+END_IF
+]]></ST>
+      </Implementation>
+    </Action>
+    <Action Name="CombineOutputs" Id="{fb280db4-19e3-45d0-a523-0567a14368b1}">
+      <Implementation>
+        <ST><![CDATA[
+// bAtState is TRUE if ALL entries are TRUE
+bAtState := TRUE;
+FOR nIndex := 1 TO nActiveMotorCount DO
+    bAtState := bAtState AND afbPositionStateMove[nIndex].bAtState;
+END_FOR
+
+// bBusy is TRUE if ANY entry is TRUE
+bBusy := FALSE;
+FOR nIndex := 1 TO nActiveMotorCount DO
+    bBusy := bBusy OR afbPositionStateMove[nIndex].bBusy;
+END_FOR
+
+// bDone is TRUE if ALL entries are TRUE
+bDone := TRUE;
+FOR nIndex := 1 TO nActiveMotorCount DO
+    bDone := bDone AND afbPositionStateMove[nIndex].bDone;
+END_FOR
+
+// bError is TRUE if ANY entry is TRUE
+// also set nShownError and increment nErrorCount
+bError := FALSE;
+nErrorCount := 0;
+FOR nIndex := 1 TO nActiveMotorCount DO
+    bError := bError OR afbPositionStateMove[nIndex].bError;
+    IF afbPositionStateMove[nIndex].bError THEN
+        nShownError := nIndex;
+        nErrorCount := nErrorCount + 1;
```

Comment:
This doesn't get out of this function block, does it? So just for debugging help after logging in?

Reply:
I'll double check to make sure this isn't a "forgot to remove" but there are a few things in here that are for debugging

Reply:
After some investigation, this is a debug parameter. It could be useful to understand if 1, 2, or 3 of your motors have errors.